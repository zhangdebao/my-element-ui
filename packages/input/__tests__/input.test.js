import Input from '../src/input.vue'
import { mount } from '@vue/test-utils' 

describe('zdb-input', () => { 
  test('input-text', () => { 
    const wrpper = mount(Input)
    expect(wrpper.html()).toContain('input type="text"')
  })

  test('input-password', () => { 
    const wrpper = mount(Input, {
      propsData: {
        type: 'password'
      }
    })
    expect(wrpper.html()).toContain('input type="password"')
  })

  test('input-password-value', () => { 
    const wrpper = mount(Input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrpper.props('value')).toBe('admin')
  })

  test('input-snapshot', () => { 
    const wrpper = mount(Input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrpper.vm.$el).toMatchSnapshot() // 第一次生成快照，存储到文件中
  })
})