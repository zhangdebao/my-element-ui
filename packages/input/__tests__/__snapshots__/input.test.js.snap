// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zdb-input input-snapshot 1`] = `
<div>
  <input
    type="password"
  />
</div>
`;
